NOTES.ANDROID

```diff
@@ -0,0 +1,66 @@
+
+ NOTES FOR THE ANDROID PLATFORMS
+ ===============================
+
+ Requirement details
+ -------------------
+
+ Beside basic tools like perl and make you'll need to download Android
+ NDK. It's available for Linux, MacOS X and Windows, but only Linux
+ version was actually tested. There is no reason to believe that MacOS X
+ wouldn't work. And as for Windows it's unclear which "shell" would work
+ [best], MSYS2 might have best chances. NDK version should play lesser
+ role, the goal is to support a range of most recent versions.
+
+ Configuration
+ -------------
+
+ Android is naturally cross-compiled target and you can't use ./config.
+ You have to use ./Configure and name your target explicitly, there are
+ android-arm, android-arm64, android-mips, android-mip64, android-x86
+ and android-x86_64. Do not pass --cross-compile-prefix, [as you might
+ be tempted,] as it will be "calculated" automatically based on chosen
+ platform. Though you'd still need to know it, the prefix, for $PATH
+ adjustment, that would allow to invoke $(CROSS_COMPILE)gcc and company.
+ [Yes, it can be summarized as "a bit messy."] Apart from PATH adjustment
+ you need to set $ANDROID_NDK environment to point at NDK directory
+ as /some/where/android-ndk-<ver>. NDK customarily supports multiple
+ Android API levels, e.g. android-14, android-21, etc. By default latest 
+ one available is chosen. If you need to target older platform, pass
+ additional -D__ANDROID_API__=N to Configure. For example, to compile for
+ ICS on ARM with NDK 10d:
+
+    ANDROID_NDK=/some/where/android-ndk-10d
+    PATH=$ANDROID_NDK/toolchains/arm-linux-androideabi-4.8/prebuild/linux-x86_64/bin:$PATH
+    [..]./Configure android-arm -D__ANDROID_API__=14
```

Comment:
???  Actually, I have *no* idea what that `[..]` is supposed to mean.  If I read it as optional, it would mean that I can choose to run `.../Configure`, and well...

Reply:
Yeah... Intention was to hint that build tree doesn't have to be same as source tree, and `...` would represent anything, such as ../openssl/Configure. But yes, I'll drop it...

Reply:
Yeah...  as it's just an example, it becomes more of a confusion than a help.  Good try, though!